### ISSUE TITLE
Multical21 | Only receive total consumption and target consumption

### ISSUE THREAD

**Issue by gitlass** [weetmuts/wmbusmeters#53]
I have a IMST871a-dongle and the water meter is a multical21(in sweden). I only receive the total consumption and target consumption values. 

I see this information in the debug:
ignoring telegram since format signature hash 0xdd34 is yet unknown

Am i doing it wrong? The total consumption value I get is correct.

Here comes the debug:
(wmbusmeters) version: 0.9.19-5-gb8bafeb
(config) using device: auto
(config) number of meters: 1
(detect) "auto" ""
(detect) driver: auto
(serialtty) opened /dev/im871a
(serial) received binary 
(im871a) are you there?
(serial /dev/im871a) sent "A5010100"
(serial) received binary A58102004CA3
(serialtty) closed /dev/im871a
(im871a) got crc16 a34c expected a34c
(im871a) on /dev/im871a
(serialtty) opened /dev/im871a
(config) meter multical21 link mode(s): c1
(config) all possible link modes that the meters might transmit on: c1
(config) listen link modes calculated to be: c1
(wmbusmeters) configured "MyTapDapper" "multical21" "63407299" encrypted
(im871a) set link mode 06
(serial /dev/im871a) sent "A5010306000206300101"
(serial) received binary A58104009CF7
(im871a) got crc16 f79c expected f79c
(im871a) set config completed
(im871a) get config
(serial /dev/im871a) sent "A5010500"
(serial) received binary A5810617FF000600B32500191000010001FF0702320001010000E0E21E
(im871a) got crc16 1ee2 expected 1ee2
(im871a) get config completed
(im871a) config: device mode 00
(im871a) config: link mode 06
(im871a) config: wmbus c-field 00
(im871a) config: wmbus mfg id 25b3 (IMS)
(im871a) config: wmbus device id 00101900
(im871a) config: wmbus version 01
(im871a) config: wmbus device type 00
(im871a) config: radio channel 01
(im871a) config: radio power level 07
(im871a) config: radio data rate 02
(im871a) config: radio rx window 32
(im871a) config: auto power saving 00
(im871a) config: auto RSSI attachment 01
(im871a) config: auto rx timestamp attachment 01
(im871a) config: led control 00
(im871a) config: rtc control 00
(config) listen to link modes: c1
(serial) waiting for stop
(serial) received binary A5E20321442D2C997240631B168D20D741524723A80788478C04E0E10912179C5C9D67FAD6DA28000000389F
(im871a) timestamp 000028da
(im871a) rssi 00
(im871a) got crc16 9f38 expected 9f38
(wmbus) received telegram 63407299 C-field=44 M-field=2c2d (KAM) A-field-version=1b A-field-dev-type=16 (Cold water meter) Ci-field=8d (Extended Link Layer II (8 Byte)) CC-field=20 (slow_resp sync) ell header ACC=d7 SN=23475241
(wmbus) frame "21442D2C997240631B168D20D741524723A80788478C04E0E10912179C5C9D67FAD6"
(wmbus) payload "A80788478C04E0E10912179C5C9D67FAD6"
(wmbus) 00: 21 length (33 bytes)
(wmbus) 01: 44 c-field (SND_NR)
(wmbus) 02: 2d2c m-field (2c2d=KAM)
(wmbus) 04: 99724063 a-field-addr (63407299)
(wmbus) 08: 1b a-field-version
(wmbus) 09: 16 a-field-type (Cold water meter)
(wmbus) 0a: 8d ci-field (Extended Link Layer II (8 Byte))
(wmbus) 0b: 20 cc-field (slow_resp sync)
(wmbus) 0c: d7 acc
(wmbus) 0d: 41524723 sn (encrypted session=1 time=3437860)
(wmbus) 21442D2C997240631B168D20D741524723
(meter) MyTapDapper: for me? 63407299
(meter) MyTapDapper: yes for me
(multical21) MyTapDapper 63407299 (Mode1) decrypting "A80788478C04E0E10912179C5C9D67FAD6"
(Mode1) IV 2D2C997240631B162041524723000000
(Mode1) decrypted first block "39977934DD45E2000053CE0800FFC708"
(Mode1) IV+1 2D2C997240631B162041524723000001
(Mode1) decrypted second block "00"
(Mode1) decrypted "39977934DD45E2000053CE0800FFC70800"
(multical21) log "telegram=|21442D2C997240631B168D20D741524723|39977934DD45E2000053CE0800FFC70800|+15"
(multical21) ignoring telegram since format signature hash 0xdd34 is yet unknown.
(multical21) 11: 3997 payload crc
(multical21) 13: 79 frame type (short frame)
(multical21) 14: 34dd format signature
(multical21) 21442D2C997240631B168D20D74152472339977934DD
{"media":"cold water","meter":"multical21","name":"MyTapDapper","id":"63407299","total_m3":0,"target_m3":0,"max_flow_m3h":0,"flow_temperature_c":127,"external_temperature_c":127,"current_status":"","time_dry":"","time_reversed":"","time_leaking":"","time_bursting":"","timestamp":"2019-12-06T11:57:28Z"}
(serial) received binary A5E20321442D2C997240631B168D20D841524723A80788478C04E0E10912179C5C9D67FAD661C600000013A9
(im871a) timestamp 0000c661
(im871a) rssi 00
(im871a) got crc16 a913 expected a913
(wmbus) received telegram 63407299 C-field=44 M-field=2c2d (KAM) A-field-version=1b A-field-dev-type=16 (Cold water meter) Ci-field=8d (Extended Link Layer II (8 Byte)) CC-field=20 (slow_resp sync) ell header ACC=d8 SN=23475241
(wmbus) frame "21442D2C997240631B168D20D841524723A80788478C04E0E10912179C5C9D67FAD6"
(wmbus) payload "A80788478C04E0E10912179C5C9D67FAD6"
(wmbus) 00: 21 length (33 bytes)
(wmbus) 01: 44 c-field (SND_NR)
(wmbus) 02: 2d2c m-field (2c2d=KAM)
(wmbus) 04: 99724063 a-field-addr (63407299)
(wmbus) 08: 1b a-field-version
(wmbus) 09: 16 a-field-type (Cold water meter)
(wmbus) 0a: 8d ci-field (Extended Link Layer II (8 Byte))
(wmbus) 0b: 20 cc-field (slow_resp sync)
(wmbus) 0c: d8 acc
(wmbus) 0d: 41524723 sn (encrypted session=1 time=3437860)
(wmbus) 21442D2C997240631B168D20D841524723
(meter) MyTapDapper: for me? 63407299
(meter) MyTapDapper: yes for me
(multical21) MyTapDapper 63407299 (Mode1) decrypting "A80788478C04E0E10912179C5C9D67FAD6"
(Mode1) IV 2D2C997240631B162041524723000000
(Mode1) decrypted first block "39977934DD45E2000053CE0800FFC708"
(Mode1) IV+1 2D2C997240631B162041524723000001
(Mode1) decrypted second block "00"
(Mode1) decrypted "39977934DD45E2000053CE0800FFC70800"
(multical21) log "telegram=|21442D2C997240631B168D20D841524723|39977934DD45E2000053CE0800FFC70800|+31"
(multical21) ignoring telegram since format signature hash 0xdd34 is yet unknown.
(multical21) 11: 3997 payload crc
(multical21) 13: 79 frame type (short frame)
(multical21) 14: 34dd format signature
(multical21) 21442D2C997240631B168D20D84152472339977934DD
{"media":"cold water","meter":"multical21","name":"MyTapDapper","id":"63407299","total_m3":0,"target_m3":0,"max_flow_m3h":0,"flow_temperature_c":127,"external_temperature_c":127,"current_status":"","time_dry":"","time_reversed":"","time_leaking":"","time_bursting":"","timestamp":"2019-12-06T11:57:44Z"}
(serial) received binary A5E20321442D2C997240631B168D20D941524723A80788478C04E0E10912179C5C9D67FAD697F70000002C0F
(im871a) timestamp 0000f797
(im871a) rssi 00
(im871a) got crc16 0f2c expected 0f2c
(wmbus) received telegram 63407299 C-field=44 M-field=2c2d (KAM) A-field-version=1b A-field-dev-type=16 (Cold water meter) Ci-field=8d (Extended Link Layer II (8 Byte)) CC-field=20 (slow_resp sync) ell header ACC=d9 SN=23475241
(wmbus) frame "21442D2C997240631B168D20D941524723A80788478C04E0E10912179C5C9D67FAD6"
(wmbus) payload "A80788478C04E0E10912179C5C9D67FAD6"
(wmbus) 00: 21 length (33 bytes)
(wmbus) 01: 44 c-field (SND_NR)
(wmbus) 02: 2d2c m-field (2c2d=KAM)
(wmbus) 04: 99724063 a-field-addr (63407299)
(wmbus) 08: 1b a-field-version
(wmbus) 09: 16 a-field-type (Cold water meter)
(wmbus) 0a: 8d ci-field (Extended Link Layer II (8 Byte))
(wmbus) 0b: 20 cc-field (slow_resp sync)
(wmbus) 0c: d9 acc
(wmbus) 0d: 41524723 sn (encrypted session=1 time=3437860)
(wmbus) 21442D2C997240631B168D20D941524723
(meter) MyTapDapper: for me? 63407299
(meter) MyTapDapper: yes for me
(multical21) MyTapDapper 63407299 (Mode1) decrypting "A80788478C04E0E10912179C5C9D67FAD6"
(Mode1) IV 2D2C997240631B162041524723000000
(Mode1) decrypted first block "39977934DD45E2000053CE0800FFC708"
(Mode1) IV+1 2D2C997240631B162041524723000001
(Mode1) decrypted second block "00"
(Mode1) decrypted "39977934DD45E2000053CE0800FFC70800"
(multical21) log "telegram=|21442D2C997240631B168D20D941524723|39977934DD45E2000053CE0800FFC70800|+47"
(multical21) ignoring telegram since format signature hash 0xdd34 is yet unknown.
(multical21) 11: 3997 payload crc
(multical21) 13: 79 frame type (short frame)
(multical21) 14: 34dd format signature
(multical21) 21442D2C997240631B168D20D94152472339977934DD
{"media":"cold water","meter":"multical21","name":"MyTapDapper","id":"63407299","total_m3":0,"target_m3":0,"max_flow_m3h":0,"flow_temperature_c":127,"external_temperature_c":127,"current_status":"","time_dry":"","time_reversed":"","time_leaking":"","time_bursting":"","timestamp":"2019-12-06T11:58:00Z"}
(serial) received binary A5E20324442D2C997240631B168D20DA70524723658873251CEF4F0612DA54744531263FDBABB320D923000000A15F
(im871a) timestamp 000023d9
(im871a) rssi 00
(im871a) got crc16 5fa1 expected 5fa1
(wmbus) received telegram 63407299 C-field=44 M-field=2c2d (KAM) A-field-version=1b A-field-dev-type=16 (Cold water meter) Ci-field=8d (Extended Link Layer II (8 Byte)) CC-field=20 (slow_resp sync) ell header ACC=da SN=23475270
(wmbus) frame "24442D2C997240631B168D20DA70524723658873251CEF4F0612DA54744531263FDBABB320"
(wmbus) payload "658873251CEF4F0612DA54744531263FDBABB320"
(wmbus) 00: 24 length (36 bytes)
(wmbus) 01: 44 c-field (SND_NR)
(wmbus) 02: 2d2c m-field (2c2d=KAM)
(wmbus) 04: 99724063 a-field-addr (63407299)
(wmbus) 08: 1b a-field-version
(wmbus) 09: 16 a-field-type (Cold water meter)
(wmbus) 0a: 8d ci-field (Extended Link Layer II (8 Byte))
(wmbus) 0b: 20 cc-field (slow_resp sync)
(wmbus) 0c: da acc
(wmbus) 0d: 70524723 sn (encrypted session=0 time=3437863)
(wmbus) 24442D2C997240631B168D20DA70524723
(meter) MyTapDapper: for me? 63407299
(meter) MyTapDapper: yes for me
(multical21) MyTapDapper 63407299 (Mode1) decrypting "658873251CEF4F0612DA54744531263FDBABB320"
(Mode1) IV 2D2C997240631B162070524723000000
(Mode1) decrypted first block "B5E27802FF200000041353CE08004413"
(Mode1) IV+1 2D2C997240631B162070524723000001
(Mode1) decrypted second block "FFC70800"
(Mode1) decrypted "B5E27802FF200000041353CE08004413FFC70800"
(multical21) log "telegram=|24442D2C997240631B168D20DA70524723|B5E27802FF200000041353CE08004413FFC70800|+63"
(dvparser) found new format "02FF2004134413" with hash dd34, remembering!
(dvparser) looking for type=instantaneous vif=Volume storagenr=0
(dvparser) match? type=instantaneous vif=Volume and storagenr=0
(dvparser) match? type=instantaneous vif=Volume and storagenr=0
(dvparser) found key 0413 for type=instantaneous vif=Volume storagenr=0
(dvparser) looking for type=unknown vif=Volume storagenr=1
(dvparser) match? type=instantaneous vif=Volume and storagenr=1
(dvparser) match? type=instantaneous vif=Volume and storagenr=1
(dvparser) match? type=instantaneous vif=Volume and storagenr=1
(dvparser) found key 4413 for type=instantaneous vif=Volume storagenr=1
(dvparser) looking for type=unknown vif=VolumeFlow storagenr=-1
(dvparser) match? type=instantaneous vif=VolumeFlow and storagenr=-1
(dvparser) match? type=instantaneous vif=VolumeFlow and storagenr=-1
(dvparser) match? type=instantaneous vif=VolumeFlow and storagenr=-1
(dvparser) looking for type=unknown vif=FlowTemperature storagenr=-1
(dvparser) match? type=instantaneous vif=FlowTemperature and storagenr=-1
(dvparser) match? type=instantaneous vif=FlowTemperature and storagenr=-1
(dvparser) match? type=instantaneous vif=FlowTemperature and storagenr=-1
(dvparser) looking for type=unknown vif=ExternalTemperature storagenr=-1
(dvparser) match? type=instantaneous vif=ExternalTemperature and storagenr=-1
(dvparser) match? type=instantaneous vif=ExternalTemperature and storagenr=-1
(dvparser) match? type=instantaneous vif=ExternalTemperature and storagenr=-1
(multical21) 11: b5e2 payload crc
(multical21) 13: 78 frame type (long frame)
(multical21) 14: 02 dif (16 Bit Integer/Binary Instantaneous value)
(multical21) 15: FF vif (Vendor extension)
(multical21) 16: 20 vife (per second)
(multical21) 17: * 0000 info codes (OK)
(multical21) 19: 04 dif (32 Bit Integer/Binary Instantaneous value)
(multical21) 1a: 13 vif (Volume l)
(multical21) 1b: * 53CE0800 total consumption (577.107000 m3)
(multical21) 1f: 44 dif (32 Bit Integer/Binary Instantaneous value storagenr=1)
(multical21) 20: 13 vif (Volume l)
(multical21) 21: * FFC70800 target consumption (575.487000 m3)
(multical21) 24442D2C997240631B168D20DA70524723B5E27802FF200000041353CE08004413FFC70800
{"media":"cold water","meter":"multical21","name":"MyTapDapper","id":"63407299","total_m3":577.107,"target_m3":575.487,"max_flow_m3h":0,"flow_temperature_c":127,"external_temperature_c":127,"current_status":"","time_dry":"","time_reversed":"","time_leaking":"","time_bursting":"","timestamp":"2019-12-06T11:58:16Z"}
^C(serial) stopping manager
(serial) event loop stopped!
(serial) closing 1 devices
(serial) no devices working
(serial) no devices working
(serialtty) closed /dev/im871a

**Comment by weetmuts** [comment]
Your meter does not transmit the temperature nor the max flow. Some Multical21 meters simply do not report the temperature, probably because they do not have this ability. 

I will probably make it so, that the json does not contain the temperature keys, if there is no temperature reported from the meter.

The " format signature hash 0xdd34 is yet unknown" is just information that wmbusmeters
has received a compressed telegram, but it cannot be decoded until you have a received 
a full not-compressed telegram. When the full telegram is received then, wmbusmeters
knows what the hash is supposed to match. So, you might have to wait for a few
compressed telegrams has arrived until you get a long telegram and then start getting real data.

I will improve the logging and the README to explain this.

I do however add hardcoded hashes to wmbusmeters, and I just added yours,
so that wmbusmeters can start decoding the compressed telegrams immediately
if it is a meter configuration type that I have seen before.

I just added your hash as well. :-)
